### PR TITLE
Update documentation of minimum weight full matching

### PR DESCRIPTION
**Pull request by fuglede**
In #4057 , we removed the requirement that the input for `minimum_weight_full_matching` be a complete bipartite graph. This simply updates the documentation of the function accordingly. In the process, we include a test showcasing the possible `ValueError` that will arise if the input graph admits no full matching.

### CODE REVIEW

File: networkx/algorithms/bipartite/matching.py [hunk 1]
```diff
@@ -475,18 +475,18 @@ def to_vertex_cover(G, matching, top_nodes=None):
 
 
 def minimum_weight_full_matching(G, top_nodes=None, weight='weight'):
-    r"""Returns the minimum weight full matching of the bipartite graph `G`.
+    r"""Returns a minimum weight full matching of the bipartite graph `G`.
```
Comment (rossbar) [inline networkx/algorithms/bipartite/matching.py:478]: Why the change here? I'm not familiar with the matching algorithms so forgive me if this is an obvious question. Are there sometimes multiple minimum-weight full-matchings for bipartite graphs?

**Comment by jarrodmillman** [comment]
@rossbar I fixed the conflict, which I believe was due to me running black.  I wanted to doublecheck with you, but I believe there is now no need for @fuglede to rebase.

**Comment by rossbar** [comment]
Good catch - I falsely attributed the merge conflict. Thanks for fixing @jarrodmillman

File: networkx/algorithms/bipartite/matching.py (hunk 1 above)
Comment (jarrodmillman) [reply networkx/algorithms/bipartite/matching.py:489]: @rossbar There can be many minimum-weight full-matchings for bipartite graphs.  A matching is an independent edge set of the graph (i.e., no two edges share a common node).  So, e.g., if you have a balanced, complete bipartite graph (each part has an equal # of nodes with no edges in each part and all possible edges between the parts) whose edges all have unit weight, then there are a large number of minimum weight matchings.

Comment (fuglede) [reply networkx/algorithms/bipartite/matching.py:489]: Yeah, that's exactly it. Say, for instance, that the partitions are `{1, 2}` and `{3, 4}`, with edges `(1, 3)`, `(1, 4)`, `(2, 3)`, `(2, 4)`, all with equal weights. Then there are two full matchings with total weight 2: `[(1, 3), (2, 4)]` and `[(1, 4), (2, 3)]`.

Really, some of the other methods in this module should be changed as well to be completely precise.

Comment (rossbar) [reply networkx/algorithms/bipartite/matching.py:489]: Thanks all for the clarification!
